src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -61,6 +61,14 @@ public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageConte
         }
 
         /// <summary>
+        /// Creates a <see cref="IIncomingPhysicalMessageContext"/> based on the current context.
+        /// </summary>
+        public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageContext(this StageConnector<ITransportReceiveContext, IIncomingPhysicalMessageContext> stageConnector, IncomingMessage incomingMessage, ITransportReceiveContext sourceContext)
```

Comment:
Agreed, there is no way for endusers to manipulate the satellite pipeline (as mentioned earlier I don't think it has to ea pipeline, `func<Task,IncomingMessage>` would be enough).

I propose we consider the satellite pipeline internal